include/wx/graphics.h

```diff
@@ -132,6 +132,49 @@ class WXDLLIMPEXP_CORE wxGraphicsObject : public wxObject
     wxDECLARE_DYNAMIC_CLASS(wxGraphicsObject);
 };
 
+// ----------------------------------------------------------------------------
+// wxGraphicsPenInfo describes a wxGraphicsPen
+// ----------------------------------------------------------------------------
+
+class wxGraphicsPenInfo : public wxPenInfo
+{
+public:
+    wxGraphicsPenInfo()
+    : wxPenInfo()
+    {
+        m_widthF = -1.0;
+    }
+
+    explicit wxGraphicsPenInfo(const wxColour& colour, double widthF = 1.0, wxPenStyle style = wxPENSTYLE_SOLID)
+    : wxPenInfo(colour, 0, style)
+    {
+        m_widthF = widthF;
+    }
+
+    // Setters for the various attributes. All of them return the object itself
+    // so that the calls to them could be chained.
+
+    wxGraphicsPenInfo& Colour(const wxColour& colour);
```

Comment:
Where is this and other methods implemented? I don't see them...

In principle, we could omit the methods present at `wxPenInfo` level. This would mean that `WidthReal()` (or whatever we end up calling it) would have to be called first, but maybe it's not such a huge limitation?

We could also use some template trickery to avoid having to provide all these methods here:
```cpp
template <class T>
class wxPenInfoBase
{
public:
   T& Colour(const wxColour& colour) { m_colour = colour; return static_cast<T&>(*this); }
   ... same thing for all the other methods ...
private:
   wxColour m_colour;
};

class wxPenInfo : public wxPenInfoBase<wxPenInfo>
{
   using wxPenInfoBase<wxPenInfo>::wxPenInfoBase;
};

class wxGraphicsPenInfo : public wxPenInfoBase<wxGraphicsPenInfo>
{
public:
   using wxPenInfoBase<wxGraphicsPenInfo>::wxPenInfoBase;

    wxGraphicsPenInfo& WidthReal(double w) { ... }
};
```

This would make it impossible to construct `wxPen` from `wxGraphicsPenInfo` too which, for me, is a bonus (because `wxPen` doesn't support floating point width).